fs/vfs/fs_unlink.c

```diff
@@ -114,13 +114,7 @@ int unlink(FAR const char *pathname)
    * or a soft link, then rm should remove the node.
    */
 
-#ifdef CONFIG_PSEUDOFS_SOFTLINKS
-  /* A soft link is the only "specal" file that we can remove via unlink(). */
-
-  if (!INODE_IS_SPECIAL(inode) || INODE_IS_SOFTLINK(inode))
-#else
   if (!INODE_IS_SPECIAL(inode))
```

Comment:
@xiaoxiang781216 FSNODEFLAG_TYPE_SOFTLINK is a type of FSNODEFLAG_TYPE_SPECIAL so the code inside the if() will not run. Did you test it with CONFIG_PSEUDOFS_SOFTLINKS ?

Reply:
@acassis, you are right. I misunderstand the logic. The change is wrong, I will abandon it.